pkg/federated/mergeResponses.go

```diff
@@ -86,24 +93,32 @@ func (d *Data) mergeMessages(msgs []string) {
 	d.Messages = append(d.Messages, msgs...)
 }
 
-// func (d *Data) mergeRelatedResults(mergedItems, newItems []SearchRelatedResult) {
-// 	klog.Info("Merge related results to federated response.")
-// 	d.writeLock.Lock()
-// 	defer d.writeLock.Unlock()
-
-// 	// Related results are grouped by kind.
-// 	for _, newItem := range newItems {
-// 		kind := newItem.Kind
-// 		for _, mergedItem := range mergedItems {
-// 			if mergedItem.Kind == kind {
-// 				// Merge the new items.
-// 				mergedItem.Count = mergedItem.Count + newItem.Count
-// 				for _, item := range newItem.Items {
-// 					item["managedHub"] = "TODO_ADD_HUB_NAME_HERE" // TODO: Add hub name to related items.
-// 					mergedItem.Items = append(mergedItem.Items, item)
-// 				}
-// 				return
-// 			}
-// 		}
-// 	}
-// }
+func (d *Data) appendRelatedResults(mergedItems, newItems []SearchRelatedResult) []SearchRelatedResult {
+	klog.V(1).Info("Merge [related] to federated response.")
+
+	if len(mergedItems) == 0 {
+		return newItems
+	}
+
+	// Related results are grouped by kind.
+	for _, newItem := range newItems {
+		found := false
+		kind := newItem.Kind
+
+		// If the kind is found, merge the newItems with the mergedItems.
+		for index, mergedItem := range mergedItems {
```

Comment:
if the search term is not drilled down, like kind:Cluster, there might be a lot of mergedItems to loop through to find `kind`.

Reply:
Agree, I'll merge this to get it working and revisit the performance after.